### PR TITLE
Generate wallet 

### PR DESCRIPTION
**Pull request by Kukks**
(typos on images have been fixed)

image

image

### CODE REVIEW

File: BTCPayServer/Controllers/StoresController.BTCLike.cs [hunk 1]
```diff
@@ -319,6 +320,38 @@ private DerivationSchemeSettings GetExistingDerivationStrategy(string cryptoCode
...
+          var client =   _ExplorerProvider.GetExplorerClient(cryptoCode);
+          var response = await client.GenerateWalletAsync(request);
+          
+          var store = HttpContext.GetStoreData();
+          var result = await AddDerivationScheme(storeId, new DerivationSchemeViewModel()
+          {
+              Confirmation = false,
+              Network = network,
+              RootFingerprint = response.AccountKeyPath.MasterFingerprint.ToString(),
+              RootKeyPath = response.AccountKeyPath.KeyPath,
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/StoresController.BTCLike.cs]: not right, should be `network.GetRootKeyPath()`

File: BTCPayServer/Controllers/StoresController.BTCLike.cs [hunk 2]
```diff
@@ -319,6 +320,38 @@ private DerivationSchemeSettings GetExistingDerivationStrategy(string cryptoCode
             return ShowAddresses(vm, strategy);
         }
 
+        [HttpPost]
+        [Route("{storeId}/derivations/{cryptoCode}/generatenbxwallet")]
+        public async Task<IActionResult> GenerateNBXWallet(string storeId, string cryptoCode, GenerateWalletRequest request)
+        {
+            var network = _NetworkProvider.GetNetwork<BTCPayNetwork>(cryptoCode);
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/StoresController.BTCLike.cs]: too much space

**Comment by britttttk** [comment]
Creating a wallet this way.. How can a user check addresses as instructed in the UI? `Please check that your BTC wallet is generating the same addresses as below.`

tACK - I'm able to send/receive fine with mnemonic seed for now, I think that more functionality on sign comes when #1218 is merged. However I don't quite understand the RPC checkbox, is that something that is to be added in another PR as well?

**Comment by Kukks** [comment]
RPC checkbox works already:
it imports any address generated to the Bitcoin Core wallet node so that
you can view your balance through the bitcoin-cli or rpc.
If you enable save private keys, you'll also have private keys imported to
the wallet and be able to spend from it.

**Comment by NicolasDorier** [comment]
`Save private keys` should be named `Is hotwallet`.  With proper warning about what it means (danger about funds being stolen, but more convenient)

**Comment by Eskyee** [comment]
Welcome to BTCPay Server ... Secure. The payment server does not need to know your private keys, so your money can't be stolen.  
that was the good old days ... lol

sorry but  have I missed something here ??? 
we are going save private keys in btcpay or NBX ??  I'm very dumbstruck here !!  
this sounds like a very very bad idea,

**Comment by Kukks** [comment]
> 
> 
> Welcome to BTCPay Server ... Secure. The payment server does not need to know your private keys, so your money can't be stolen.
> that was the good old days ... lol
> 
> sorry but have I missed something here ???
> we are going save private keys in btcpay or NBX ?? I'm very dumbstruck here !!
> this sounds like a very very bad idea,

Opt-in feature for specific use-cases. Just ignore it if you don't need it.

**Comment by Eskyee** [comment]
Opt-in feature for specific use-cases !! yes ok, that's much better for use-cases, I understand a lot more now,  Is a hotwallet, and not the store main xpub wallet, is that correct ??  could you also explain how secure this new method would be, as  a hot wallet. thanks guys
sorry for my rants!! ;)

**Comment by pavlenex** [comment]
A hot wallet is needed mostly for developers. Nothing will change and the method of adding xpub key remains the same, as Kukks said, this just extend the use-cases and is completely optional and advanced feature. Lightning is a hot wallet as well by the way. It won't influence you if you're not using it if that's your question.

**Comment by Eskyee** [comment]
yes now I`m very happy, just got myself confused.. as a extra opt-in feature it's a kool idea, @pavlenex

**Comment by Kukks** [comment]
Updated to this
image

**Comment by Kukks** [comment]
can we merge now @NicolasDorier

**Pull request merged**